Extract evidence for the given term in the publication (#9)
sensein/EviSense: README.md

evisense-cli extract-evidence --config config.yml --source paper.pdf --terms '["Astrocyte","Ependymal"]'

# Using comma-separated values
evisense-cli extract-evidence --config config.yml --source paper.pdf --terms "Astrocyte,Ependymal"
```

3. Directory of PDFs:
```bash
evisense-cli extract-evidence --config config.yml --source path/to/papers/ --terms "Neuron"
```

4. Raw Text Input:
```bash
# Short text
evisense-cli extract-evidence --config config.yml --source "This is a neuroscience text..." --terms "Neuron"
```

### Configuration Options

- `provider`: (Optional) Specifies which provider to use ("all" or specific provider name)
- `default`: Default provider to use in these scenarios:
  1. When no provider is specified
  2. When the specified provider fails to initialize
  3. When authentication fails (e.g., invalid/missing API keys)
  4. When all selected providers return errors
- For each provider:
  - `base_url`: API endpoint
  - `api_key`: Authentication key (if required)
  - `models`: List of models to use
  - `default_model`: Default model if no specific models listed

### Example Configuration

```yaml
llm:
  default: "ollama"  # Specifies the default LLM provider

>   provider: "openrouter" #select provider to use, e.g., ollama or openrouter. the value all will make use of all the providers specified in config in current case ollama and openrouter

Reviewer: i  don't understand the comment

Reviewer 2: might be
```suggestion
  provider: "openrouter" # Select provider to use, e.g., ollama or openrouter. 
                         # All values of providers specified in config will be used, in current case ollama and openrouter
```
?

  ollama:
    base_url: "http://localhost:11434"
    default_model: "deepseek-r1:14b"  # Default model for Ollama
    models:
      - "deepseek-r1:14b"
      - "qwen2.5-coder:14b"

  openrouter:
    api_key: "sk-or-v1"
    base_url: "https://openrouter.ai/api/v1"
    default_model: "gpt-4"  # Default model for OpenRouter
    models:
      - "gpt-4"
      - "gpt-4-turbo"

  grobid_server_url: "http://localhost:8070" #optional, if not specified, uses the default one http://http://localhost:8070

> ```

Reviewer: I don't understand the format of the configuration file. It seems to me as a combination of providing possible options: e.g., `ollama` and `openrouter` setting the options. Not sure what is `default` vs `provider` under `llm`.
 Similarly, not sure how to interpret `models` vs `default_model`, since there is no option to chose it via cli, I thought that this is the place to choose.

Author: @djarecka provider is a different service provider, like `openrouter`, which you can use to access the commercial llm models like gpt. `ollama` is the open-source alternative of the `openrouter` which you can download and run locally or in a self-hosted environment. `default_model` is a default fall-back model. `models` are all the models that will be used. If you just want to run using a single model, say, `gpt-4` you don't need `models` parameter. The idea behind `models` is that I can run the same prompt using multiple models and compare the response. This is also something that I plan to use in evaluation in a manner similar to `majority voting` in ensemble models.

Author: Regarding no option to chose via cli, you need to provide a path to the config file.  I will update the readme to make it clear.

Reviewer: yes, this is clear to me, that you provide `config`, but I still didn't why you asked for `default_model` and `models`. If by `default_model` you mean "fall-back model", perhaps the key should be changed. The description should also say which one are mandatory etc.

And I still don't understand the keys for `llm`. If I just want to use `ollama`, do I have to set `provider` as well to `ollama`?
